### PR TITLE
Upgrade network API version to 2021-08-01

### PR DESCRIPTION
**Pull request by lodrem**
#### What type of PR is this?

/kind cleanup
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azureclients/interfaceclient/interface.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ package interfaceclient
 import (
 	"context"
 
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-08-01/network"
 
 	"sigs.k8s.io/cloud-provider-azure/pkg/retry"
 )
```
Comment (feiskyer) [inline pkg/azureclients/interfaceclient/interface.go:25]: Please also update the `APIVersion` below for each client

Comment (feiskyer) [reply pkg/azureclients/interfaceclient/interface.go:25]: `APIVersion` is defined in each client's `interface.go`.

Comment (lodrem) [reply pkg/azureclients/interfaceclient/interface.go:25]: updated.

**Comment by feiskyer** [comment]
@lodrem could you cherrypick this to 1.0~1.24 release branches?

**Pull request merged**